pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,23 +19,50 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "file"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
 func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
 	var err error
 	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, err = az.getStorageAccounts()
-		if err != nil {
-			// TODO: create a storage account and container
-			return "", "", err
+	if len(storageAccount) == 0 {
+		// find a storage account that matches storageType
+		accounts, err = az.getStorageAccounts(storageType)
+		if err != nil || len(accounts) == 0 {
+			uniqueID := az.controllerCommon.resourceGroup + az.controllerCommon.location + az.controllerCommon.subscriptionID
```

Comment:
this reveals too much (perhaps sensitive) information

Reply:
@rootfs there is hash like function `MakeCRC32` which would transform `uniqueID` to a hash string, e,g. if `uniqueID` is `resourcegroupwestuscc1624c7-3f1d-4ed3-a855-668a86e96ad8`, it will convert to `2380111856`, you could see no one could parse string `2380111856` to get info like resourcegroup, subscriptionID. 
The reason I use both `uniqueID` and UUID is that it's unique for every subscriptionID, resourcegroup, thus the account name conflict possibility be near zero as 1/10 000 000 000 in one resourcegroup. If that possibility happens, user could delete the PVC and recreate PVC again.

Reply:
where is that unique string transformed into a CRC string?

Reply:
Suppose using uuid is enough, though we need do substr to fit for the limit constraint.
Ref:
https://github.com/Azure/azure-quickstart-templates/blob/master/101-storage-account-create/azuredeploy.json#L20
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-string#uniquestring